### ISSUE TITLE
It may be possible to run sslyze in parallel mode again

### ISSUE THREAD

**Issue by jsf9k** [18F/domain-scan#188]
I confirmed that [nassl](https://github.com/nabla-c0d3/nassl) (a dependency of [sslyze](https://github.com/nabla-c0d3/sslyze)) fixed some serious memory leaks as of version 1.0.2 (c. December 16, 2017).  

As a result, it may now be possible to again run sslyze in parallel mode.  This should be tested.

**Comment by konklone** [comment]
> As a result, it may now be possible to again run sslyze in parallel mode. This should be tested.

The memory leaks I identified were from running `sslyze` in serial mode. I actually never observed them in parallel mode, which I found pretty strange, but consistent.

The issues running `sslyze` in parallel mode were much more gnarly, and related to deadlocks around shared mutexes caused by multithreading. (And in Lambda mode, the `multiprocessing` functions needed to do that are specifically disabled.)
